cpp/ycm/IdentifierUtils.cpp

```diff
@@ -114,38 +113,34 @@ const char *const NOT_FOUND = "YCMFOOBAR_NOT_FOUND";
 FiletypeIdentifierMap ExtractIdentifiersFromTagsFile(
   const fs::path &path_to_tag_file ) {
   FiletypeIdentifierMap filetype_identifier_map;
-  std::string tags_file_contents;
+  std::vector< std::string > tags_file_contents;
 
   try {
     tags_file_contents = ReadUtf8File( path_to_tag_file );
   } catch ( ... ) {
     return filetype_identifier_map;
```

Comment:
You should be able to write `return {};` (C++11) to default construct the result.
This allows you to move the line `FiletypeIdentifierMap filetype_identifier_map;` further down in front of the loop.